src/nitypes/bintime/_datetime.py

```diff
@@ -0,0 +1,286 @@
+
+    This class only supports the UTC time zone and does not support timezone-naive times.
+
+    This class does not support the ``fold`` property for disambiguating repeated times for daylight
+    saving time and time zone changes.
+    """
+
+    min: ClassVar[DateTime]
+    """The earliest supported :any:`DateTime` object, midnight on Jan 1, 0001, UTC."""
+
+    max: ClassVar[DateTime]
+    """The latest supported :any:`DateTime` object, 11:59pm on Dec 31, 9999, UTC."""
+
+    __slots__ = ["_offset"]
+
+    _offset: TimeDelta
+
+    @overload
+    def __init__(self) -> None: ...  # noqa: D107 - missing docstring in __init__
+
+    @overload
+    def __init__(  # noqa: D107 - missing docstring in __init__
+        self, value: _OtherDateTime, /
+    ) -> None: ...
+
+    def __init__(
+        self,
+        year: SupportsIndex | _OtherDateTime | None = None,
+    ) -> None:
+        """Initialize an DateTime."""
+        self._offset = self.__class__._to_offset(year)
+
+    @singledispatchmethod
+    @classmethod
+    def _to_offset(cls, value: object) -> TimeDelta:
+        raise invalid_arg_type("value", "datetime", value)
+
+    @_to_offset.register
+    @classmethod
+    def _(cls, value: ht.datetime) -> TimeDelta:
+        if value.tzinfo != dt.timezone.utc:
+            raise invalid_arg_value("value.tzinfo", "datetime.timezone.utc", value.tzinfo)
+        return TimeDelta(value - _HT_EPOCH_1904)
+
+    @_to_offset.register
+    @classmethod
+    def _(cls, value: dt.datetime) -> TimeDelta:
+        if value.tzinfo != dt.timezone.utc:
+            raise invalid_arg_value("value.tzinfo", "datetime.timezone.utc", value.tzinfo)
+        return TimeDelta(value - _DT_EPOCH_1904)
+
+    @_to_offset.register
+    @classmethod
+    def _(cls, value: None) -> TimeDelta:
+        return TimeDelta()
+
+    @classmethod
+    def from_ticks(cls, ticks: SupportsIndex) -> Self:
+        """Create an DateTime from a 128-bit fixed point number expressed as an integer."""
+        self = cls.__new__(cls)
+        self._offset = TimeDelta.from_ticks(ticks)
+        return self
+
+    @classmethod
+    def from_offset(cls, offset: TimeDelta) -> Self:
+        """Create an DateTime from a TimeValue offset from the epoch, Jan 1, 1904."""
+        self = cls.__new__(cls)
+        self._offset = offset
+        return self
+
+    def _to_datetime_datetime(self) -> dt.datetime:
+        """Return self as a :any:`datetime.datetime`."""
+        return _DT_EPOCH_1904 + self._offset._to_datetime_timedelta()
+
+    def _to_hightime_datetime(self) -> ht.datetime:
+        """Return self as a :any:`hightime.datetime`."""
+        return _HT_EPOCH_1904 + self._offset._to_hightime_timedelta()
+
+    # Calculating the year/month/day requires knowledge of leap years, days per month, etc., so
+    # defer to hightime.datetime.
+    @property
+    def year(self) -> int:
+        """The year."""
+        return self._to_hightime_datetime().year
```

Comment:
This class leans pretty heavily on hightime.datetime, so it really ought to cache the converted value.